lib/api_auth/base.rb

```diff
@@ -78,23 +80,31 @@ def request_too_old?(headers)
       end
     end
 
-    def signatures_match?(headers, secret_key)
+    def signatures_match?(headers, secret_key, options)
+      options = options.merge(:with_http_method => true)
       if match_data = parse_auth_header(headers.authorization_header)
         hmac = match_data[2]
-        hmac == hmac_signature(headers, secret_key)
+        return (
```

Comment:
I was just trying to make is really clear that there is a logical OR here, do you think it is more clear without the parens? or i could make local variables for each hmac_signature case and compare against each of them for shorter line length.